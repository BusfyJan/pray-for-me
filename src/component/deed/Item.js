import React from 'react'
import PropTypes from 'prop-types'
import Badge from '@material-ui/core/Badge'
import styled from 'styled-components'

import DeedIcon from 'component/deed/Icon.js'

const BadgeStyled = styled(Badge)`
  & > span {
    width: 18px;
    height: 18px;
  }
`

const Item = ({ onClick, count, type }) => (
  <span
    onClick={() => {
      if (onClick) {
        onClick()
      }
    }}
  >
    <BadgeStyled badgeContent={count} color="primary">
      <DeedIcon type={type} />
    </BadgeStyled>
  </span>
)

Item.propTypes = {
  onClick: PropTypes.func,
  count: PropTypes.number.isRequired,
  type: PropTypes.string.isRequired
}

export default Item
