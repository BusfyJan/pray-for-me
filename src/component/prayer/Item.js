import React, { Component } from 'react'
import styled from 'styled-components'
import moment from 'moment'
import PropTypes from 'prop-types'

import Card, { CardHeader, CardContent, CardActions } from 'material-ui/Card'
import { FormattedMessage } from 'react-intl'
import Button from 'material-ui/Button'
import AddIcon from 'material-ui-icons/Add'
import DoneIcon from 'material-ui-icons/Done'
import Divider from 'material-ui/Divider'
import { green, blue, orange } from 'material-ui/colors'

import CircularProgress from 'component/util/CircularProgress.js'
import PrayerIcon from 'component/prayer/Icon.js'
import PrayerTitle from 'component/prayer/Title.js'
import PrayerDescription from 'component/prayer/Description.js'
import DeedsList from 'component/deed/List.js'

const CardStyled = styled(Card)`
  margin-bottom: 25px;
`

const ActionsWrapper = styled.div`
  width: 100%;
  text-align: right;
  margin-right: 5px;
`

const OwnRequestInfoWrapper = styled.span`
  font-weight: bold;
  color: white;
  padding: 3px 5px;
  margin-left: 5px;
  background: ${orange[500]};
  border-radius: 3px;
  font-size: 0.75em;
`

const TimestampInfoWrapper = styled.span`
  font-weight: bold;
  color: white;
  padding: 3px 5px;
  margin-left: 5px;
  background: ${blue[500]};
  border-radius: 3px;
  font-size: 0.75em;
`

const UserStatusOnlineInfoWrapper = styled.span`
  font-weight: bold;
  color: white;
  padding: 3px 5px;
  margin-left: 5px;
  background: ${green[500]};
  border-radius: 3px;
  font-size: 0.75em;
`

class Item extends Component {
  constructor() {
    super()

    this.state = {
      isClosing: false
    }
  }

  render() {
    const { data, onResponseRequest, onCloseRequest } = this.props
    return (
      <CardStyled>
        <CardHeader
          avatar={<PrayerIcon type={data.type} />}
          title={
            <span>
              <PrayerTitle type={data.type} />
              {data.isMine ? (
                <OwnRequestInfoWrapper>
                  <FormattedMessage
                    id="component.prayer.item.yours"
                    defaultMessage="yours"
                  />
                </OwnRequestInfoWrapper>
              ) : data.isUserOnline ? (
                <UserStatusOnlineInfoWrapper>
                  <FormattedMessage
                    id="component.prayer.item.online"
                    defaultMessage="online"
                  />
                </UserStatusOnlineInfoWrapper>
              ) : null}
              <TimestampInfoWrapper>
                {moment(-parseInt(data.timestamp, 10))
                  .locale('sk')
                  .fromNow()}
              </TimestampInfoWrapper>
            </span>
          }
          subheader={<PrayerDescription type={data.type} />}
        />
        <Divider />
        <CardContent>
          <DeedsList items={data.deeds} />
        </CardContent>
        <Divider />
        <CardActions disableActionSpacing>
          <ActionsWrapper>
            {data.isMine ? (
              <Button
                onClick={() => {
                  if (this.state.isClosing) {
                    return
                  }

                  this.setState({ isClosing: true }, () => {
                    onCloseRequest().catch(() => {
                      this.setState({
                        isClosing: false
                      })
                    })
                  })
                }}
                raised
                dense
                color="primary"
              >
                <FormattedMessage
                  id="component.prayer.item.closePrayerRequest"
                  defaultMessage="Close prayer request"
                />
                &nbsp;
                {this.state.isClosing ? (
                  <CircularProgress size={20} thickness={7} color="white" />
                ) : (
                  <DoneIcon />
                )}
              </Button>
            ) : (
              <Button
                onClick={() => {
                  onResponseRequest()
                }}
                raised
                dense
                color="primary"
              >
                <FormattedMessage
                  id="component.prayer.item.addDeedButton"
                  defaultMessage="Add deed"
                />
                &nbsp;
                <AddIcon />
              </Button>
            )}
          </ActionsWrapper>
        </CardActions>
      </CardStyled>
    )
  }
}

Item.propTypes = {
  data: PropTypes.object.isRequired,
  onResponseRequest: PropTypes.func.isRequired,
  onCloseRequest: PropTypes.func.isRequired
}

export default Item
