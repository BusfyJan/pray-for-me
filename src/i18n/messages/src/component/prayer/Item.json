[
  {
    "id": "component.prayer.item.addDeedButton",
    "defaultMessage": "Add deed"
  }
]