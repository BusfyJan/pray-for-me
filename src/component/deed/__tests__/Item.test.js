import React from 'react'
import Item from '../Item'
import renderer from 'react-test-renderer'

const mockCallBack = jest.fn()

test('Item render', () => {
  const component = renderer.create(
    <Item type="mass" count={5} onClick={mockCallBack} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  tree.props.onClick()

  expect(mockCallBack).toBeCalled()
})
