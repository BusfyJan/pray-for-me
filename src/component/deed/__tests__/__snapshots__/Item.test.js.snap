// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item render 1`] = `
<span
  onClick={[Function]}
>
  <span
    className="MuiBadge-root-1 sc-bdVaJa fRAYjp"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-6 sc-bwzfXH jlGedI"
      color={undefined}
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
    <span
      className="MuiBadge-badge-2 MuiBadge-colorPrimary-3"
    >
      5
    </span>
  </span>
</span>
`;
